substrate/client/consensus/babe/src/lib.rs

```diff
@@ -896,18 +909,10 @@ where
 	}
 }
 
-/// Extract the BABE pre digest from the given header. Pre-runtime digests are
-/// mandatory, the function will return `Err` if none is found.
-pub fn find_pre_digest<B: BlockT>(header: &B::Header) -> Result<PreDigest, Error<B>> {
-	// genesis block doesn't contain a pre digest so let's generate a
-	// dummy one to not break any invariants in the rest of the code
-	if header.number().is_zero() {
-		return Ok(PreDigest::SecondaryPlain(SecondaryPlainPreDigest {
-			slot: 0.into(),
-			authority_index: 0,
-		}))
-	}
-
+/// Extract the BABE pre digest from the given header. If the pre-runtime
+/// digest is not present (e.g. genesis or aura -> babe migration), the
+/// function will return `Ok(None)`.
+pub fn find_pre_digest<B: BlockT>(header: &B::Header) -> Result<Option<PreDigest>, Error<B>> {
```

Comment:
This should be changed to return `Option`, `Result` is not that great (was before, I know).

There is only one place in `verify` where we would need to check if the digest exists multiple times.